dl/src/main/scala/com/intel/analytics/bigdl/optim/SGD.scala

```diff
@@ -32,17 +32,19 @@ class SGD[@specialized(Float, Double) T: ClassTag](implicit ev: TensorNumeric[T]
     config: Table, state: Table = null): (Tensor[T], Array[T]) = {
 
     val _state = if (state == null) config else state
-    val lrSchedule = config.get[LearningRateSchedule]("learningRateSchedule").getOrElse(Default())
-    lrSchedule.updateHyperParameter(config, _state)
-
-    val wd = config.get[Double]("weightDecay").getOrElse(0.0)
-    val mom = config.get[Double]("momentum").getOrElse(0.0)
-    val damp = config.get[Double]("dampening").getOrElse(mom)
-    val nesterov = config.get[Boolean]("nesterov").getOrElse(false)
-    val lrs = config.get[Tensor[T]]("learningRates").getOrElse(null)
-    val wds = config.get[Tensor[T]]("weightDecays").getOrElse(null)
-
-    require(!nesterov || (mom > 0 && damp == 0),
+    val scheduler = config.getOrElse[HyperParameterScheduler]("hyperParameterScheduler", Default())
+    scheduler.updateHyperParameter(config, _state)
+
```

Comment:
If we only read hyperparameter from clr, cweightDecay..., then when the user are writing their user-defined HyperParameterScheduler, he needs to write more codes to define cweightDecay, clearningRateDecay... even he doesn't change the weightDecay and learningRateDecay.